code/game/objects/structures/fusioncell.dm

```diff
@@ -0,0 +1,95 @@
+/obj/structure/fusioncell
+	name = "Fusion Cell"
+	desc = "A simple way to store radioactive energy with many capabilities."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "oldshieldoff"
+	density = FALSE
+	anchored = FALSE
+	max_integrity = INFINITY
+	var/emitting = FALSE
+	var/radiation_count = 0
+	var/emittingmount = 0
+	var/i
+	var/datum/looping_sound/geiger/soundloop
+
+
+	/obj/structure/fusioncell/attackby(obj/item/I, mob/user, params)
+		if(I.tool_behaviour == TOOL_WRENCH)
+			default_unfasten_wrench(user, I, time = 5)
+		return
+
+	/obj/structure/fusioncell/proc/update_sound()
+		var/datum/looping_sound/geiger/loop = soundloop
+		if(emitting == FALSE)
+			loop.stop()
+			return
+		if(radiation_count < 3500)
+			loop.stop()
+			return
+		loop.start()
+
+	/obj/structure/fusioncell/interact(mob/user)
+		if(emitting == FALSE)
+			to_chat(user, "You turn the cell on")
+			rad_insulation = RAD_FULL_INSULATION //prevents infinite rads ie cant emit rads and gain rads at the same time
+			icon_state = "oldshieldon"
+			update_icon()
+			emitting = TRUE
+			update_sound()
+			emittingmount = radiation_count/100
+			for(i = 0, 101>i, i++)
+				if(emitting == FALSE)
+					return
+				if(radiation_count < 3500)
+					visible_message("<span class='warning'>The [src] does not have enough power to continue!!!.</span>",null,null,5)
+					return
+				visible_message("<span class='warning'>The [src] starts to release a pulse of <b>[emittingmount]</b> rads!!!.</span>",null,null,5)
+				sleep(20 * world.tick_lag)
+				radiation_pulse(src,emittingmount,5,TRUE,TRUE)
+				radiation_count -= emittingmount
+				visible_message("<span class='warning'>The [src] released a pulse of <b>[emittingmount]</b> rads!!!.</span>",null,null,5)
+				continue
+			return
```

Comment:
What is this for?

Reply:
whenever it didn't compile I added returns until it worked

Reply:
*cry